pkg/daemon/update.go

```diff
@@ -131,8 +143,18 @@ func (dn *Daemon) reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) *stri
 	// we don't currently configure groups or users in place. we can't fix it if
 	// something changed here.
 	if !reflect.DeepEqual(oldIgn.Passwd, newIgn.Passwd) {
-		msg := "Ignition passwd section contains changes"
-		return &msg
+		// check if the prior config is empty and that this is the first time running.
+		// if so, the SSHKey from the cluster config and user "core" must be added to machine config,.
+		// emptyMC := mcfgv1.MachineConfig{}
+		// if reflect.DeepEqual(emptyMC.Spec.Config.Passwd, oldIgn.Passwd) && oldConfig.Kind == "" {
+		// 	glog.Infof("Reconcilable: Initializing ssh keys")
+		if len(newIgn.Passwd.Users[0].SSHAuthorizedKeys) > 0 {
+			glog.Info("SSH Keys reconcilable")
```

Comment:
Hmm, I think we need to be even stronger here, no? There's lots of other changes that could've snuck into there that aren't reconcilable. For the time being, if the only thing we want to support is SSH key reconciliation for a single user, we can do something like this:

1. rule out group changes by doing `DeepEqual` on `Passwd.Groups`
2. check that `len(Passwd.Users)` on both are the same and equal to 1
3. make *copies* of the old and new `Passwd.Users[0]` and zero out the `SSHAuthorizedKeys` elements and then `DeepEqual` those instead. That way, SSH key changes don't interfere with the `DeepEqual` check.

Not sure if there's a cleaner way of doing this short of busting out golang reflect and checking every field.